presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorAnalyzeTableMetadata.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.connector;
+
+import com.facebook.presto.spi.ConnectorTableHandle;
+import com.facebook.presto.spi.ConnectorTableMetadata;
+import com.facebook.presto.spi.statistics.TableStatisticsMetadata;
+
+import static java.util.Objects.requireNonNull;
+
+public class ConnectorAnalyzeTableMetadata
+{
+    private final TableStatisticsMetadata statisticsMetadata;
+    private final ConnectorTableMetadata tableMetadata;
+    private final ConnectorTableHandle tableHandle;
+
+    public ConnectorAnalyzeTableMetadata(TableStatisticsMetadata statisticsMetadata, ConnectorTableMetadata tableMetadata, ConnectorTableHandle tableHandle)
+    {
+        this.statisticsMetadata = requireNonNull(statisticsMetadata, "statisticsMetadata is null");
+        this.tableMetadata = requireNonNull(tableMetadata, "tableMetadata is null");
+        this.tableHandle = requireNonNull(tableHandle, "tableHandle is null");
+    }
+
+    public TableStatisticsMetadata getStatisticsMetadata()
+    {
+        return statisticsMetadata;
+    }
+
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    public ConnectorTableHandle getTableHandle()
```

Comment:
We need the `getAnalyzeTableMetadata()` to return an updated table handle which has dependency on the analyze properties (e.g. what range of table to analyze). It will later be passed to `getTableLayouts()`